integration/query_comparision_test.go

```diff
@@ -986,8 +986,84 @@ func TestQueryComparisonLte(t *testing.T) {
 	}
 }
 
+func TestQueryComparisonNin(t *testing.T) {
+	t.Parallel()
+	providers := []shareddata.Provider{shareddata.Scalars, shareddata.Composites}
+	ctx, collection := setup(t, providers...)
+
+	var scalarDataTypesFilter bson.A
+	for _, scalarDataType := range shareddata.Scalars.Docs() {
+		scalarDataTypesFilter = append(scalarDataTypesFilter, scalarDataType.Map()["value"])
+	}
+
+	var compositeDataTypesFilter bson.A
+	for _, compositeDataType := range shareddata.Composites.Docs() {
+		compositeDataTypesFilter = append(compositeDataTypesFilter, compositeDataType.Map()["value"])
+	}
+
+	for name, tc := range map[string]struct {
+		filter      bson.D
+		expectedIDs []any
+		err         mongo.CommandError
+	}{
+		"ForScalarDataTypes": {
+			filter:      bson.D{{"value", bson.D{{"$nin", scalarDataTypesFilter}}}},
+			expectedIDs: []any{"array-empty", "document", "document-composite", "document-empty"},
+		},
+		"ForCompositeDataTypes": {
+			filter: bson.D{{"value", bson.D{{"$nin", compositeDataTypesFilter}}}},
+			expectedIDs: []any{
+				"binary", "binary-empty",
+				"bool-false", "bool-true",
+				"datetime", "datetime-epoch", "datetime-year-max", "datetime-year-min",
+				"double", "double-max", "double-nan", "double-negative-infinity", "double-negative-zero",
+				"double-positive-infinity", "double-smallest", "double-whole", "double-zero",
+				"int32", "int32-max", "int32-min", "int32-zero",
+				"int64", "int64-max", "int64-min", "int64-zero",
+				"null",
+				"objectid", "objectid-empty",
+				"regex", "regex-empty",
+				"string", "string-double", "string-empty", "string-whole",
+				"timestamp", "timestamp-i",
+			},
+		},
+
+		"NilInsteadOfArray": {
+			filter: bson.D{{"value", bson.D{{"$nin", nil}}}},
+			err: mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: `$nin needs an array`,
+			},
+		},
+		"StringInsteadOfArray": {
+			filter: bson.D{{"value", bson.D{{"$nin", "foo"}}}},
+			err: mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: `$nin needs an array`,
+			},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var actual []bson.D
```

Comment:
tbd: actual is used 5 lines later -> move and add empty line

Reply:
fixed